apps/juicebox_web/assets/js/reactions/recorder.js

```diff
@@ -12,86 +13,110 @@ class ReactionRecorder extends Component {
     super();
     this.webcam = new Webcam();
     this.state = {
-      webcamConnected: false,
-      recording: null,
+      isConnected: false,
+      isRecording: false,
+      video: null,
+      stream: null,
     }
   }
 
+  startButton = () =>
+    <button onClick={this.startWebcam}>Record a reaction</button>
+
+  stopButton = () =>
+    <button onClick={this.stopWebcam}>&times;</button>
+
+  recordButton = () =>
+    <button onClick={this.record}>Record</button>
+
+  clearButton = () =>
+    <button onClick={this.clearRecording}>Clear</button>
+
+  sendButton = () =>
+    <button onClick={this.send}>Send</button>
+
+
   render() {
-    const { webcamConnected, recording } = this.state;
+    const {
+      isConnected,
+      video,
+      stream,
+      isRecording,
+    } = this.state;
 
     return (
-      <div>
-        <video ref={(video) => this.video = video} loop></video>
-        { webcamConnected ? (
-            <button onClick={this.stopWebcam}>Disconnect</button>
-        ) : (
-            <button onClick={this.startWebcam}>Record a reaction</button>
-        )}
-
-        { (webcamConnected && !recording) && <button onClick={this.record}>Record</button> }
-        { (webcamConnected && recording) && <button onClick={this.clearRecording}>Clear</button> }
-        { recording && <button onClick={this.send}>Send</button> }
+      <div className="reaction-recorder">
+        <div className={`webcam
+          ${!isConnected ? 'offline' : ''}
+          ${isRecording ? 'recording' : ''}`}>
+
+          { show(this.stopButton).when(isConnected) }
+          <video src={video || stream} loop autoPlay></video>
+
+          <div className="recorder-controls">
+            { show(this.startButton).when(!isConnected) }
```

Comment:
Good point, especially considering I was going to allow `when` to accept a predicate function!